### PR TITLE
Update Canal and Calico VXLAN to v3.22.2 and remove IPVS restrictions

### PR DESCRIPTION
**Pull request by xmudrii**
**What type of PR is this?**

/kind feature

**What this PR does / why we need it**:

This PR updates Canal and Calico VXLAN to v3.22.2. This allows users to use kube-proxy in IPVS mode on AMD64 clusters running Kubernetes 1.23 and newer.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
xref #2010 

**Special notes for your reviewer**:

* We don't have restrictions for Canal/Calico VXLAN on ARM clusters with Kubernetes 1.23 and newer. This can be added in a follow up
* We need to cherry-pick this PR to the release/v1.4 branch

**Does this PR introduce a user-facing change?**:
```release-note
This PR updates Canal and Calico VXLAN addons to v3.22.2. This allows users to use kube-proxy in IPVS mode on AMD64 clusters running Kubernetes 1.23 and newer. It currently remains impossible to use kube-proxy in IPVS mode on ARM64 clusters running Kubernetes 1.23 and newer.
```

### CODE REVIEW

File: pkg/templates/images/images.go [hunk 1]
```diff
@@ -165,9 +165,9 @@ func FindResource(name string) (Resource, error) {
 
 func baseResources() map[Resource]map[string]string {
 	return map[Resource]map[string]string{
-		CalicoCNI:         {"*": "docker.io/calico/cni:v3.22.0"},
-		CalicoController:  {"*": "docker.io/calico/kube-controllers:v3.22.0"},
-		CalicoNode:        {"*": "docker.io/calico/node:v3.22.0"},
+		CalicoCNI:         {"*": "docker.io/calico/cni:v3.22.2"},
+		CalicoController:  {"*": "docker.io/calico/kube-controllers:v3.22.2"},
+		CalicoNode:        {"*": "docker.io/calico/node:v3.22.2"},
```
Comment (ahmedwaleedmalik) [inline pkg/templates/images/images.go:170]: Not really relevant to this PR but maybe we should switch to `quay.io` for these images. They maintain the same images on docker hub and quay.io 

https://github.com/projectcalico/calico/blob/master/.semaphore/push-images/cni-plugin.yml#L43

Comment (kron4eg) [reply pkg/templates/images/images.go:170]: @ahmedwaleedmalik send patch!

**Pull request merged**